@Neumann-A @dg0yt  @Cheney-W @LilyWangLL 
What happens to the CI?

```
Clearing contents of D:\buildtrees
Failure to remove_all_inside("D:\buildtrees") due to file "\\?\D:\buildtrees\ffmpeg": The process cannot access the file because it is being used by another process.
##[error]vcpkg clean failed
At C:\a\1\s\scripts\azure-pipelines\test-modified-ports.ps1:133 char:5
+     throw "vcpkg clean failed"
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : OperationStopped: (vcpkg clean failed:String) [], RuntimeException
+ FullyQualifiedErrorId : vcpkg clean failed
##[error]PowerShell exited with code '1'.
```

--- a/ports/qt5-base/patches/fix_angle.patch
+++ b/ports/qt5-base/patches/fix_angle.patch
@@ -1,0 +1,23 @@
+diff --git a/src/plugins/platforms/windows/qwindowseglcontext.cpp b/src/plugins/platforms/windows/qwindowseglcontext.cpp
+index 4ae087dfa..a0c3060a0 100644
+--- a/src/plugins/platforms/windows/qwindowseglcontext.cpp
++++ b/src/plugins/platforms/windows/qwindowseglcontext.cpp
+@@ -46,6 +46,9 @@
+ 
+ #if defined(QT_OPENGL_ES_2_ANGLE) || defined(QT_OPENGL_DYNAMIC)
+ #  include <EGL/eglext.h>
++#ifndef EGL_ANGLE_platform_angle
++#  include <EGL/eglext_angle.h>
++#endif
+ #endif
+ 
+ QT_BEGIN_NAMESPACE
+@@ -210,7 +213,7 @@ bool QWindowsEGLStaticContext::initializeAngle(QWindowsOpenGLTester::Renderers p
+             { EGL_PLATFORM_ANGLE_TYPE_ANGLE, EGL_PLATFORM_ANGLE_TYPE_D3D11_ANGLE, EGL_NONE },
+             { EGL_PLATFORM_ANGLE_TYPE_ANGLE, EGL_PLATFORM_ANGLE_TYPE_D3D9_ANGLE, EGL_NONE },
+             { EGL_PLATFORM_ANGLE_TYPE_ANGLE, EGL_PLATFORM_ANGLE_TYPE_D3D11_ANGLE,
+-              EGL_PLATFORM_ANGLE_DEVICE_TYPE_ANGLE, EGL_PLATFORM_ANGLE_DEVICE_TYPE_WARP_ANGLE, EGL_NONE }
++              EGL_PLATFORM_ANGLE_DEVICE_TYPE_ANGLE, EGL_PLATFORM_ANGLE_DEVICE_TYPE_D3D_WARP_ANGLE, EGL_NONE }
+         };
+         const EGLint *attributes = nullptr;
+         if (preferredType & QWindowsOpenGLTester::AngleRendererD3d11)

--- a/ports/qt5-base/patches/fix_angle.patch
+++ b/ports/qt5-base/patches/fix_angle.patch
@@ -1,18 +1,39 @@
 diff --git a/src/plugins/platforms/windows/qwindowseglcontext.cpp b/src/plugins/platforms/windows/qwindowseglcontext.cpp
-index 4ae087dfa..a0c3060a0 100644
+index 4ae087dfa..06603faab 100644
 --- a/src/plugins/platforms/windows/qwindowseglcontext.cpp
 +++ b/src/plugins/platforms/windows/qwindowseglcontext.cpp
-@@ -46,6 +46,9 @@
+@@ -46,6 +46,30 @@
  
  #if defined(QT_OPENGL_ES_2_ANGLE) || defined(QT_OPENGL_DYNAMIC)
  #  include <EGL/eglext.h>
++
 +#ifndef EGL_ANGLE_platform_angle
-+#  include <EGL/eglext_angle.h>
-+#endif
++#define EGL_ANGLE_platform_angle 1
++#define EGL_PLATFORM_ANGLE_ANGLE          0x3202
++#define EGL_PLATFORM_ANGLE_TYPE_ANGLE     0x3203
++#define EGL_PLATFORM_ANGLE_MAX_VERSION_MAJOR_ANGLE 0x3204
++#define EGL_PLATFORM_ANGLE_MAX_VERSION_MINOR_ANGLE 0x3205
++#define EGL_PLATFORM_ANGLE_TYPE_DEFAULT_ANGLE 0x3206
++#define EGL_PLATFORM_ANGLE_DEBUG_LAYERS_ENABLED_ANGLE 0x3451
++#define EGL_PLATFORM_ANGLE_DEVICE_TYPE_ANGLE 0x3209
++#define EGL_PLATFORM_ANGLE_DEVICE_TYPE_HARDWARE_ANGLE 0x320A
++#define EGL_PLATFORM_ANGLE_DEVICE_TYPE_NULL_ANGLE 0x345E
++#define EGL_PLATFORM_ANGLE_NATIVE_PLATFORM_TYPE_ANGLE 0x348F
++#endif /* EGL_ANGLE_platform_angle */
++
++#ifndef EGL_ANGLE_platform_angle_d3d
++#define EGL_ANGLE_platform_angle_d3d 1
++#define EGL_PLATFORM_ANGLE_TYPE_D3D9_ANGLE 0x3207
++#define EGL_PLATFORM_ANGLE_TYPE_D3D11_ANGLE 0x3208
++#define EGL_PLATFORM_ANGLE_DEVICE_TYPE_D3D_WARP_ANGLE 0x320B
++#define EGL_PLATFORM_ANGLE_DEVICE_TYPE_D3D_REFERENCE_ANGLE 0x320C
++#define EGL_PLATFORM_ANGLE_ENABLE_AUTOMATIC_TRIM_ANGLE 0x320F
++#endif /* EGL_ANGLE_platform_angle_d3d */
++
  #endif
  
  QT_BEGIN_NAMESPACE
-@@ -210,7 +213,7 @@ bool QWindowsEGLStaticContext::initializeAngle(QWindowsOpenGLTester::Renderers p
+@@ -210,7 +234,7 @@ bool QWindowsEGLStaticContext::initializeAngle(QWindowsOpenGLTester::Renderers p
              { EGL_PLATFORM_ANGLE_TYPE_ANGLE, EGL_PLATFORM_ANGLE_TYPE_D3D11_ANGLE, EGL_NONE },
              { EGL_PLATFORM_ANGLE_TYPE_ANGLE, EGL_PLATFORM_ANGLE_TYPE_D3D9_ANGLE, EGL_NONE },
              { EGL_PLATFORM_ANGLE_TYPE_ANGLE, EGL_PLATFORM_ANGLE_TYPE_D3D11_ANGLE,

--- a/ports/qt5-base/portfile.cmake
+++ b/ports/qt5-base/portfile.cmake
@@ -67,6 +67,7 @@ qt_download_submodule(  OUT_SOURCE_PATH SOURCE_PATH
                             patches/Qt5PluginTarget.patch
                             patches/create_cmake.patch
                             patches/Qt5GuiConfigExtras.patch   # Patches the library search behavior for EGL since angle is not build with Qt
+                            patches/fix_angle.patch            # Failed to create OpenGL context for format QSurfaceFormat ...
                     )
 
 # Remove vendored dependencies to ensure they are not picked up by the build

--- a/ports/qt5-base/vcpkg.json
+++ b/ports/qt5-base/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "qt5-base",
   "version": "5.15.7",
-  "port-version": 3,
+  "port-version": 4,
   "description": "Qt5 Application Framework Base Module. Includes Core, GUI, Widgets, Networking, SQL, Concurrent and other essential qt components.",
   "homepage": "https://www.qt.io/",
   "license": null,

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6114,7 +6114,7 @@
     },
     "qt5-base": {
       "baseline": "5.15.7",
-      "port-version": 3
+      "port-version": 4
     },
     "qt5-canvas3d": {
       "baseline": "0",

--- a/versions/q-/qt5-base.json
+++ b/versions/q-/qt5-base.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "92139eef98d34113c6a6a6d254a7c60898cd0a4d",
+      "version": "5.15.7",
+      "port-version": 4
+    },
+    {
       "git-tree": "81ce731dc0595354a8eb77ea9557f4b7a6ca44d3",
       "version": "5.15.7",
       "port-version": 3

